src/events.ts

```diff
@@ -590,59 +592,39 @@ export class ContractCreatedEvent extends BlockchainEvent {
       renter,
       duration,
       acceptanceType,
-      revocationType,
+      renterCanRevoke,
       rentFee,
       renterCancellationFee,
       renteeCancellationFee,
     ] = event.data
 
-    const parsedDuration = duration.toString() !== DurationAction.Infinite && JSON.parse(duration.toString())
-    const isDurationFixed = parsedDuration && Boolean(parsedDuration.fixed)
-    const isDurationSubscription = parsedDuration && Boolean(parsedDuration.subscription)
+    const parsedDuration = JSON.parse(duration.toString())
     const parsedAcceptance = JSON.parse(acceptanceType.toString())
     const isAutoAcceptance = Boolean(parsedAcceptance.autoAcceptance === null || parsedAcceptance.autoAcceptance)
-    const isrevocationTypeNoRevocation = Boolean(revocationType.toString() === RevocationAction.NoRevocation)
-    const isrevocationTypeOnSubscriptionChange = Boolean(
-      revocationType.toString() === RevocationAction.OnSubscriptionChange,
-    )
     const parsedRentFee = JSON.parse(rentFee.toString())
     const isRentFeeToken = Boolean(parsedRentFee.tokens)
-    const parsedRenterCancellationFee = renterCancellationFee.toString() && JSON.parse(renterCancellationFee.toString())
-    const isRenterCancellationFeeFixed = Boolean(parsedRenterCancellationFee && parsedRenterCancellationFee.fixedTokens)
-    const isRenterCancellationFeeFlexible = Boolean(
-      parsedRenterCancellationFee && parsedRenterCancellationFee.flexibleTokens,
-    )
-    const isRenterCancellationFeeNft = Boolean(parsedRenterCancellationFee && parsedRenterCancellationFee.nft >= 0)
-    const parsedRenteeCancellationFee = renteeCancellationFee.toString() && JSON.parse(renteeCancellationFee.toString())
-    const isRenteeCancellationFeeFixed = Boolean(parsedRenteeCancellationFee && parsedRenteeCancellationFee.fixedTokens)
-    const isRenteeCancellationFeeFlexible = Boolean(
-      parsedRenteeCancellationFee && parsedRenteeCancellationFee.flexibleTokens,
-    )
-    const isRenteeCancellationFeeNft = Boolean(parsedRenteeCancellationFee && parsedRenteeCancellationFee.nft >= 0)
+    const parsedRenterCancellationFee =
+      renterCancellationFee.toString() !== CancellationFeeAction.None && JSON.parse(renterCancellationFee.toString())
+    const isRenterCancellationFeeFixed = Boolean(parsedRenterCancellationFee.fixedTokens)
```

Comment:
Instead of creating several flags, we can use a switch like so:

```
switch (true) {
  case CancellationFeeAction.FixedTokens in renterCancellationFee:
    this.renterCancellationFeeType = CancellationFeeAction.FixedTokens
    this.renterCancellationFee = bnToBn(parsedRenterCancellationFee.fixedTokens).toString()
    this.renterCancellationFeeRounded = roundBalance(this.renterCancellationFee)
    break
  case CancellationFeeAction.FlexibleTokens in renterCancellationFee:
    this.renterCancellationFeeType = CancellationFeeAction.FlexibleTokens
    this.renterCancellationFee = bnToBn(parsedRenterCancellationFee.flexibleTokens).toString()
    this.renterCancellationFeeRounded = roundBalance(this.renterCancellationFee)
    break
  case CancellationFeeAction.NFT in renterCancellationFee:
    this.renterCancellationFeeType = CancellationFeeAction.NFT
    this.renterCancellationFee = Number(parsedRenterCancellationFee.nft)
    this.renterCancellationFeeRounded = this.renterCancellationFee
    break
  default:
    this.renterCancellationFeeType = CancellationFeeAction.None
    this.renterCancellationFee = null
    this.renterCancellationFeeRounded = null
    break
}
```